VideoLocker/src/main/java/org/edx/mobile/view/EditUserProfileFragment.java

```diff
@@ -412,24 +411,9 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                 final Uri imageUri = CropImageActivity.getImageUriFromResult(data);
                 final Rect cropRect = CropImageActivity.getCropRectFromResult(data);
                 if (null != imageUri && null != cropRect) {
-                    executePhotoTask(new SetAccountImageTask(getActivity(), username, imageUri, cropRect) {
-                        @Override
-                        protected void onSuccess(Void aVoid) throws Exception {
-                            hideLoading();
-                        }
-
-                        @Override
-                        protected void onException(Exception e) throws RuntimeException {
-                            super.onException(e);
-                            hideLoading();
-                        }
-
-                        private void hideLoading() {
-                            if (null != viewHolder) {
-                                viewHolder.profileImageProgress.setVisibility(View.GONE);
-                            }
-                        }
-                    });
+                    final Task task = new SetAccountImageTask(getActivity(), username, imageUri, cropRect);
```

Comment:
I actually did exactly that initially, but the problem is that `SetAccountImageTask` also performs the cropping prior to sending the image to the server. This is also the desired behaviour, since the cropping should be performed in a background worker thread, and the whole operation should also be considered to be one task for cancellation purposes (as per the current dynamics of having the update and delete tasks interrupt each other). This is probably another of those situations that could be solved by chaining tasks with RxJava.